### PR TITLE
Ci/improve serving

### PR DESCRIPTION
**Pull request by juliendenize**
## Description
Improve servings.

Add docker
- [x] Adding Dockerfile
- [x] Adding test + push latest when push on main
- [x] Addng push for releases

Adding media_dir for serving
- [x] Pydantic settings
- [x] Dataset builders (if relevant)
- [x] Unit tests for folders
- [ ] Unit tests for S3 (Probably for a next PR)

### CODE REVIEW

**Comment by juliendenize** [comment]
@benlabbe @BertrandRenault 
In this PR I made significant changes for serving by adding the media_dir which would benefit users in my opinion to allow them to point to a data warehouse instead of having to painfully duplicate their data after each dataset generation.

It makes the view urls a bit weirder for the users especially for the folderbuilder as they might have to give a urls_relative_path pointing to the parent of source_dir.

Could you give me feedbacks ?

A comprehensive review is welcomed, I already put some stuff to later be able to work with S3 storage but I didn't test them as I need to understand better localstack.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Test
+        run: |
+          mkdir -p library
+          mkdir -p library/fake_dataset
+          touch library/fake_dataset/info.json
+          echo '{"id": "fake_dataset", "name": "Fake Dataset", "description": "A fake dataset for testing purposes"}' > library/fake_dataset/info.json
+          docker run -p 8000:8000 --name pixano -d -v /home/data ${{ env.TEST_TAG }}
+          sleep 5
+          EXCPECTED_RESPONSE='[{\"id\":\"fake_dataset\",\"name\":\"Fake Dataset\",\"description\":\"A fake dataset for testing purposes\"}]'
+          RESPONSE=$(curl http://localhost:8000)
```
Comment (BertrandRenault) [inline .github/workflows/docker.yml:49]: url should be http://localhost:8000/datasets/info

File: pixano/app/settings.py [hunk 2]
```diff
@@ -6,87 +6,103 @@
...
             Used if library_dir is an S3 path
         aws_access_key: S3 AWS access key. Used if library_dir is an S3 path
         aws_secret_key: S3 AWS secret key. Used if library_dir is an S3 path
-        local_model_dir: Local path to models. Used if library_dir is an S3 path
-        data_dir: Dataset directory as Path | S3Path
-        model_dir: Model directory as Path
     """
 
-    library_dir: str = (Path.cwd() / "library").as_posix()
+    library_dir: Path | S3Path = (Path.cwd() / "library").as_posix()
```
Comment (BertrandRenault) [inline pixano/app/settings.py:35]: remove .as_posix(), it's used to get a string

File: tests/app/test_settings.py [hunk 3]
```diff
@@ -14,17 +14,19 @@
...
+        assert settings.models_dir == Path("/home/user/model")
 
         # Test 2 all undefined:
         settings = Settings()
-        assert settings.library_dir == (Path.cwd() / "library").as_posix()
-        assert settings.data_dir == Path(settings.library_dir)
-        assert settings.model_dir == settings.data_dir / "models"
+        assert settings.library_dir == Path.cwd() / "library"
+        assert settings.media_dir == Path.cwd() / "media"
+        assert settings.models_dir == settings.library_dir / "models"
```
Comment (BertrandRenault) [inline tests/app/test_settings.py:29]: before, models_dir has been set to /home/dir/model (without 's'). Choose "model" or "models" with consistency

**Review by BertrandRenault — APPROVED** [review]

**Pull request merged**